syntax = "proto3";

message PciDeviceState {
  message MsiXTableEntry {
    uint64  address   = 1;
    uint32  data      = 2;
    uint32  control   = 3;
  }

  uint32  bus                           = 1;
  uint32  device                        = 2;
  uint32  function                      = 3;
  bytes   config_space                  = 4;
  repeated MsiXTableEntry msix_entries  = 5;
  bool    pcie                          = 6;
}
